debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerConnection.java

```diff
@@ -219,12 +246,35 @@ public void getChangesForTables(SqlServerChangeTable[] changeTables, Lsn interva
 
         int idx = 0;
         for (SqlServerChangeTable changeTable : changeTables) {
-            final String query = GET_ALL_CHANGES_FOR_TABLE.replace(STATEMENTS_PLACEHOLDER, changeTable.getCaptureInstance());
+            String query = GET_ALL_CHANGES_FOR_TABLE.replace(STATEMENTS_PLACEHOLDER, changeTable.getCaptureInstance());
```

Comment:
This should not be at two different places but in a reused method. Also as this is on critical path I think the build statements should be cached to avoid performance hits.

Reply:
No doubt about it! I'm going to change it to avoid creating the query each time